compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/NodeCloner.java

```diff
@@ -969,6 +970,14 @@ public void visit(BLangGroupExpr source) {
         clone.typedescType = source.typedescType;
     }
 
+    @Override
+    public void visit(BLangLetExpression source) {
+        BLangLetExpression clone = new BLangLetExpression();
+        source.cloneRef = clone;
+        clone.letVarDeclarations = source.letVarDeclarations;
```

Comment:
Please see `cloneList` method

Reply:
Seems like cloning letvarDeclarations is missed. 
Cannot use cloneList since BLangLetVariable is not a Node, 
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/21209/commits/98c1e0005e3e22d3fc21862dcd5d569b801a8509